src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,862 @@
+                editor.InsertParameter(containingMethod, insertionIndex, parameter);
+            }
+
+            return editor.GetChangedRoot();
+        }
+
+        /// <summary>
+        /// This method iterates through the variables in the expression and maps the variables back to the parameter
+        /// it is associated with. It then maps the parameter back to the argument at the invocation site and gets the
+        /// index to retrieve the updated arguments at the invocation.
+        /// </summary>
+        private TExpressionSyntax CreateNewArgumentExpression(SyntaxEditor editor, ISyntaxFactsService syntaxFacts,
+            Dictionary<TIdentifierNameSyntax, IParameterSymbol> mappingDictionary,
+            IEnumerable<TIdentifierNameSyntax> variables,
+            ImmutableDictionary<IParameterSymbol, int> parameterToArgumentMap,
+            SeparatedSyntaxList<SyntaxNode> updatedInvocationArguments)
+        {
+            foreach (var variable in variables)
+            {
+                if (mappingDictionary.TryGetValue(variable, out var mappedParameter))
+                {
+                    var parameterMapped = false;
+                    if (parameterToArgumentMap.TryGetValue(mappedParameter, out var index))
+                    {
+                        var updatedInvocationArgument = updatedInvocationArguments.ToArray()[index];
+                        var argumentExpression = syntaxFacts.GetExpressionOfArgument(updatedInvocationArgument);
+                        var parenthesizedArgumentExpression = editor.Generator.AddParentheses(argumentExpression, includeElasticTrivia: false);
+                        editor.ReplaceNode(variable, parenthesizedArgumentExpression);
+                        parameterMapped = true;
+                    }
+
+                    if (mappedParameter.HasExplicitDefaultValue && !parameterMapped)
+                    {
+                        var generatedExpression = GenerateExpressionFromOptionalParameter(mappedParameter);
+                        var parenthesizedGeneratedExpression = editor.Generator.AddParentheses(generatedExpression, includeElasticTrivia: false);
+                        editor.ReplaceNode(variable, parenthesizedGeneratedExpression);
+                    }
+                }
+            }
+
+            return (TExpressionSyntax)editor.GetChangedRoot();
+        }
+
+        private static ImmutableDictionary<IParameterSymbol, int> MapParameterToArgumentsAtInvocation(
+            ISemanticFactsService semanticFacts, SeparatedSyntaxList<SyntaxNode> arguments,
+            SemanticModel invocationSemanticModel, CancellationToken cancellationToken)
+        {
+            var mapping = ImmutableDictionary.CreateBuilder<IParameterSymbol, int>();
+            for (var i = 0; i < arguments.Count; i++)
+            {
+                var argumentParameter = semanticFacts.FindParameterForArgument(invocationSemanticModel, arguments.ToArray()[i], cancellationToken);
```

Comment:
```suggestion
                var argumentParameter = semanticFacts.FindParameterForArgument(invocationSemanticModel, arguments[i], cancellationToken);
```

Reply:
didn't know you could index into separatedsyntaxlists, I was wondering why there was no get method